### DISCUSSION TITLE
Who uses aqua?

### DISCUSSION THREAD

**Discussion by suzuki-shunsuke** [aquaproj/aqua#2124] [Deprecated (General)]
This discussion was inspired to https://github.com/spotbugs/spotbugs/discussions/1607 .
We really appreciate it if you share projects or companies using aqua and how they use aqua.

Your sharing would motivate us.
Thanks in advance.

## OSS (Public Repositories)

https://github.com/aquaproj/user-list

## Companies

- @masterpointio

**Comment by Gowiem** [comment]
Stumbled upon aqua through a comment @sheldonhull made in the SweetOps slack. Since then I've been excitedly rolling out aqua across all our customers / projects at @masterpointio so we can avoid having version mismatches and do much simpler installs. I'm really enjoying it and think it's a great project. Keep up the good work!

**Reply by suzuki-shunsuke** [reply]
Thank you for sharing!
I joined SweetOps Slack. https://sweetops.slack.com/
I'm really happy to hear that you are enjoying aqua and rolling out aqua across your customers and projects!

**Comment by suzuki-shunsuke** [comment]
[tfaction](https://github.com/suzuki-shunsuke/tfaction) uses aqua heavily to install various tools such as Terraform, tfsec, tflint, tfcmt, and so on. aqua enables to install tools easily, and to manage tool versions and update tools per working directory.

I use aqua for my various OSS such as [tfcmt](https://github.com/suzuki-shunsuke/tfcmt), [tfaction](https://github.com/suzuki-shunsuke/tfaction), [github-comment](https://github.com/suzuki-shunsuke/github-comment), and so on.
aqua enables contributors to install tools easily and prevent version mismatches.
And we can verify issues of CI on our laptop because we can install tools same versions with CI.
I enable aqua's checksum verification to prevent tools from being tampered.

I also introduced aqua to companies I belong to then I could clean up shell scripts installing tools and update them continuously by Renovate.

**Comment by ponkio-o** [comment]
Our company (Retty Inc.) is also uses aqua in some projects. Especially, our team (internally called platform team) heavily uses it that used to manage tools for infrastructure operations and application deployment.
aqua is very useful because many tools can be quickly installed on our team member's laptops and versions can be continuously updated with Reanovate.

Thank you for development this great tool👍👍

**Comment by sheldonhull** [comment]
Well I've been lurking around a while and am a big fan :-) 

I've used on some of the open source projects from my company and use for all my internal projects.
I've found it great for using in CI + local to simplify things, and have recommended it wherever I can.

The one area that I think it shines the most for me despite it being a small subset of it's probably intentional usage is the global config. 90% of my development tools are controlled via aqua so setting up a codespace or new laptop is insanely easy.

I combine this with `chezmoi` so whenever I need to update any machine with my dotfiles, I just run `chezmoi apply` and aqua automatically installs upgraded packages for me. 


Keep up the great work and I hope the community continues to grow. Fantastic design so far and really forward thinking.

Been keeping notes from myself on this here [notes](https://notes.sheldonhull.com/development/tooling/aqua/) and will try to help improve the docs where I can soon.

**Comment by NikitaCOEUR** [comment]
Hello,

I just gave a presentation about the tool to my company, and I've started deploying it to some repositories.

Originally, I was looking for a straightforward way to make SOPS (https://github.com/getsops/sops) available to our developers. Aqua made perfect sense, so I added its installation via Puppet and our Ansible playbooks, eliminating the need to hassle with a separate deployment tool, as Aqua takes care of it.

Furthermore, I regularly discover new tools with this registry that's extremely rich in packages! (I just made my first pull request for https://github.com/evilmartians/lefthook because I'm bringing it along too!)

Thank you very much for the work you do; I've been following you for a few months (discovered with tfcmt).
See you.

**Reply by suzuki-shunsuke** [reply]
Thank you for your comment!
We're really glad to hear that you spread aqua to your company!

**Comment by ponkio-o** [comment]
Hi,

The [SWET](https://engineering.dena.com/team/quality/swet/) team at [DeNA Co., Ltd.](https://dena.com/intl/), of which I'm a member, also uses aqua extensively. Our team manages the company-wide GitHub Actions self-hosted runner, and aqua supports CI/CD for the various repositories needed to manage the self-hosted runner.
aqua's "strict version control" and "security" conscious design are excellent, and it has become an indispensable tool for our development.

Thank you for developing a great OSS.

**Comment by akitok** [comment]
Hi! I'm @akitok from [CADDi Inc.](https://caddi.com/)
"aqua" is very cool OSS! We are using aqua as our standard tool to set up GitHub Actions runners.

Then, I sent a https://github.com/aquaproj/aqua/pull/2626
Thank you as always!

**Comment by istone-you** [comment]
Hello,

At our company, [microCMS Co., Ltd.](https://microcms.co.jp/), we also use “aqua” in our local development environment and GitHub Actions!

It’s an incredibly useful tool that always supports our workflow.
Thank you for providing such a great tool!
